src/Adapter/MSTest.TestAdapter/MSTestSettings.cs

```diff
@@ -831,4 +906,162 @@ private static void SetGlobalSettings(
+        //  },
+        //  "parallelism" : {
+        //      "enabled": true/false,
+        //      "workers": positive int,
+        //      "scope": method/class,
+        //  },
+        //  ... remaining settings
+        // }
+        MSTestSettings settings = new();
+
+        ParseBooleanSetting(configuration, "captureTraceOutput", logger, value => settings.CaptureDebugTraces = value);
+        ParseBooleanSetting(configuration, "enableBaseClassTestMethodsFromOtherAssemblies", logger, value => settings.EnableBaseClassTestMethodsFromOtherAssemblies = value);
+        ParseBooleanSetting(configuration, "forcedLegacyMode", logger, value => settings.ForcedLegacyMode = value);
+        ParseBooleanSetting(configuration, "mapInconclusiveToFailed", logger, value => settings.MapInconclusiveToFailed = value);
+        ParseBooleanSetting(configuration, "mapNotRunnableToFailed", logger, value => settings.MapNotRunnableToFailed = value);
+        ParseBooleanSetting(configuration, "treatDiscoveryWarningsAsErrors", logger, value => settings.TreatDiscoveryWarningsAsErrors = value);
+        ParseBooleanSetting(configuration, "considerEmptyDataSourceAsInconclusive", logger, value => settings.ConsiderEmptyDataSourceAsInconclusive = value);
+        ParseBooleanSetting(configuration, "treatClassAndAssemblyCleanupWarningsAsErrors", logger, value => settings.TreatClassAndAssemblyCleanupWarningsAsErrors = value);
+        ParseBooleanSetting(configuration, "considerFixturesAsSpecialTests", logger, value => settings.ConsiderFixturesAsSpecialTests = value);
+        ParseBooleanSetting(configuration, "cooperativeCancellationTimeout", logger, value => settings.CooperativeCancellationTimeout = value);
+        ParseBooleanSetting(configuration, "orderTestsByNameInClass", logger, value => settings.OrderTestsByNameInClass = value);
+        ParseBooleanSetting(configuration, "parallelism:enabled", logger, value => settings.OrderTestsByNameInClass = value);
+
+        ParseIntegerSetting(configuration, "timeout:test", logger, value => settings.TestTimeout = value);
+        ParseIntegerSetting(configuration, "timeout:assemblyCleanup", logger, value => settings.AssemblyCleanupTimeout = value);
+        ParseIntegerSetting(configuration, "timeout:assemblyInitialize", logger, value => settings.AssemblyInitializeTimeout = value);
+        ParseIntegerSetting(configuration, "timeout:classInitialize", logger, value => settings.ClassInitializeTimeout = value);
+        ParseIntegerSetting(configuration, "timeout:classCleanup", logger, value => settings.ClassCleanupTimeout = value);
+        ParseIntegerSetting(configuration, "timeout:testInitialize", logger, value => settings.TestInitializeTimeout = value);
+        ParseIntegerSetting(configuration, "timeout:testCleanup", logger, value => settings.TestCleanupTimeout = value);
+
+        ParseStringSetting(configuration, "settingsFile", logger, fileName => settings.TestSettingsFile = fileName);
```

Comment:
I think we also don't want to support this one. We are in the new world so there is no point in allowing to load an old setting file